src/main.js

```diff
@@ -80,8 +104,62 @@ const findDoubleWinner = ([p1Name, p1a, p1b], [p2Name, p2a, p2b]) => {
 
 }
 
+// check that players own the correct tokens
+const checkTokenType = (tokenList, tokenType, playerLedger, playerName) => {
+  let allTokensOwned = true
+  tokenList.map( t => {
+    if (!playerLedger || !playerLedger[tokenType].includes(t)) {
+      console.log(`${playerName} must own at least one "${t}" token to play.`)
+      const boughtToken = buyToken(playerName, t, tokenType, tokenLedger)
+      if (boughtToken === true) { return }
+      allTokensOwned = false
+    }
+  })
+  return allTokensOwned
+}
+
+const checkTokens = (game, format, gameComponents, mode, playerName, tokenLedger) => {
+  let allTokensOwned = true
+  const playerLedger = tokenLedger[playerName]
+  let componentsCheck = checkTokenType(gameComponents, "components", playerLedger, playerName)
+  let formatCheck = checkTokenType([format], "formats", playerLedger, playerName)
+  let gameCheck = checkTokenType([game], "games", playerLedger, playerName)
+  let modeCheck = checkTokenType([mode], "modes", playerLedger, playerName)
+  if (componentsCheck === false || formatCheck === false || gameCheck === false || modeCheck === false) {
+    allTokensOwned = false
+  }
+  jsonfile.writeFileSync("tokens.json", tokenLedger)
```

Comment:
Is this write needed? I don't see where `tokenLedger` might have changed through these checks.

Reply:
I just figured we might as well wait until all the buyToken loops run and then just rewrite this file once (buyToken happens inside the checks).

Reply:
Ah I see. I think all of these operations could be pulled apart a little more in that case. Checking, buying, and saving  could be their own independent operations and then something could use all of those in order to check what they own, ask if they want to buy, then save. 

`checkTokens` should probably return a manifest of what elements are owned and what are not, then the calling process can decide what to do with that information.

Reply:
gave it a shot